doc/userguide/lua/libs/bytevar.rst

```diff
@@ -0,0 +1,76 @@
+Bytevar
+#######
+
+The ``suricata.bytevar`` module provides access to variables defined by 
+``byte_extract`` and ``byte_math`` keywords in Suricata rules. 
+
+It is only available in Suricata Lua rules, not output scripts.
+
+Setup
+*****
+
+::
+
+    local bytevars = require("suricata.bytevar")
+
+Module Functions
+****************
+
+.. function:: bytevars.ensure(sig, varname)
+
+   Ensures that the ``bytevar`` exists and sets it up for further use
+   in the script. Must be called during ``init()``.
+
+   :param sig: The signature object passed to ``init()``
+   :param string varname: Name of the variable as defined in the rule
+
+   :raises error: If the variable name is unknown
+   :raises error: If too many byte variables are registered
+
+   Example:
+
+   ::
+
+       function init(sig)
+           bytevars.ensure(sig, "var1")
+           bytevars.ensure(sig, "var2")
+           return {}
+       end
+
```

Comment:
This is different than how we've been documenting Lua functions, but it works quite well.. Example render:

![image](https://github.com/user-attachments/assets/a9093ffb-d6f8-41fd-b776-1926daa28e79)

Reply:
I like it